### PR TITLE
Enable spi for the stm32U585 device

### PR DESCRIPTION
**Pull request by FRASTM**
This PR defines the SPI peripheral of the stm32U5xx soc device and enables SPI1 instance for the b_u585i_iot02a
SPI2 and SPI3 nodes are also defined.

The rcc configuration is changed to accept lower spi frequency with APB2 prescaler set to 2 so that PCLK2 is HCLK/2.
-->  PASS the testcase spi_loopback on SPI1 when pins D11 = D12 are connected on the CN13.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: boards/arm/b_u585i_iot02a/Kconfig.defconfig [hunk 1]
```diff
@@ -8,4 +8,13 @@ if BOARD_B_U585I_IOT02A
...
 	default "b_u585i_iot02a"
 
+config SPI_STM32_INTERRUPT
+	default y
+	depends on SPI
+
+choice STM32_LPTIM_CLOCK
+	default STM32_LPTIM_CLOCK_LSE
+	depends on STM32_LPTIM_TIMER
+endchoice
```
Comment (erwango) [inline boards/arm/b_u585i_iot02a/Kconfig.defconfig:18]: Not related. Please remove

File: boards/arm/nucleo_l073rz/arduino_r3_connector.dtsi [hunk 2]
```diff
@@ -35,5 +35,4 @@
 	};
 };
 
-arduino_i2c: &i2c1 {};
```
Comment (erwango) [inline boards/arm/nucleo_l073rz/arduino_r3_connector.dtsi:38]: Not related

File: boards/arm/b_u585i_iot02a/b_u585i_iot02a.yaml [hunk 3]
```diff
@@ -8,3 +8,6 @@ toolchain:
   - xtools
 ram: 786
 flash: 2048
+supported:
+  - arduino_spi
```
Comment (erwango) [inline boards/arm/b_u585i_iot02a/b_u585i_iot02a.yaml:12]: Please add adruino spi to board's 'arduino_r3_connector.dtsi' file

**Comment by FRASTM** [comment]
$ west build -p auto -b b_u585i_iot02a tests/drivers/spi/spi_loopback

```
Running test suite test_spi
=============================================================
START - test_spi_loopback
I: SPI test on buffers TX/RX 0x20000129/0x200007fa
I: SPI test slow config
I: Start complete multiple
I: CS control inhibited (no GPIO device)
I: Passed
I: Start complete loop
I: Passed
I: Start null tx
I: Passed
I: Start half start
I: Passed
I: Start half end
I: Passed
I: Start every 4
I: Passed
I: Start async call
I: Passed
I: SPI test fast config
I: Start complete multiple
I: CS control inhibited (no GPIO device)
I: Passed
I: Start complete loop
I: Passed
I: Start null tx
I: Passed
I: Start half start
I: Passed
I: Start half end
I: Passed
I: Start every 4
I: Passed
I: Start async call
I: Passed
I: Start complete loop
I: CS control inhibited (no GPIO device)
I: Passed
I: Start complete loop
I: CS control inhibited (no GPIO device)
I: Passed
I: All tx/rx passed
 PASS - test_spi_loopback in 0.76 seconds
=============================================================
Test suite test_spi succeeded
=============================================================
PROJECT EXECUTION SUCCESSFUL
```

File: boards/arm/b_u585i_iot02a/doc/index.rst [hunk 4]
```diff
@@ -197,6 +199,7 @@ Default Zephyr Peripheral Mapping:
 - LD1 : PH7
 - LD2 : PH6
 - user button : PC13
+- SPI1 NSS/SCK/MISO/MOSI : PE12/P13/P14/P15 (Arduino SPI)
```
Comment (erwango) [inline boards/arm/b_u585i_iot02a/doc/index.rst:202]: s/NSS/CS

File: tests/drivers/spi/spi_loopback/boards/b_u585i_iot02a.conf [hunk 5]
```diff
@@ -0,0 +1,5 @@
+CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_1"
+CONFIG_SPI_STM32_INTERRUPT=y
```
Comment (erwango) [inline tests/drivers/spi/spi_loopback/boards/b_u585i_iot02a.conf:2]: Not needed since done at board level.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/b_u585i_iot02a/doc/index.rst (hunk 4 above)
Comment (FRASTM) [reply boards/arm/b_u585i_iot02a/doc/index.rst:202]: pin D10 is named SPI1_NSS

File: tests/drivers/spi/spi_loopback/boards/b_u585i_iot02a.conf (hunk 5 above)
Comment (FRASTM) [reply tests/drivers/spi/spi_loopback/boards/b_u585i_iot02a.conf]: removed from the .conf at test levem

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_l073rz/arduino_r3_connector.dtsi (hunk 2 above)
Comment (FRASTM) [reply boards/arm/nucleo_l073rz/arduino_r3_connector.dtsi]: done

**Review by MaureenHelm — APPROVED** [review]

File: boards/arm/b_u585i_iot02a/Kconfig.defconfig (hunk 1 above)
Comment (erwango) [reply boards/arm/b_u585i_iot02a/Kconfig.defconfig:18]: Still there.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/b_u585i_iot02a/doc/index.rst (hunk 4 above)
Comment (FRASTM) [reply boards/arm/b_u585i_iot02a/doc/index.rst:202]: the board DTS has now spi& pinctrl of 
```
    <&spi1_nss_pe12 spi1_sck_pe13
    &spi1_miso_pe14 &spi1_mosi_pe15>;
```

File: boards/arm/b_u585i_iot02a/Kconfig.defconfig (hunk 1 above)
Comment (FRASTM) [reply boards/arm/b_u585i_iot02a/Kconfig.defconfig]: removed

**Review by erwango — APPROVED** [review]

**Pull request merged**